flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,204 @@
+
+                    self.enable_logger(logger=logger)
+
+                    self.on_log = PublishMQTT.on_log_cb
+                    self.on_publish = PublishMQTT.on_publish_cb
+                    self.on_disconnect = PublishMQTT.on_disconnect_cb
+
+                    #Handle SSL/TLS communication w/out certificate authentication
+                    if not self.config.get('certificates',{}).get('client_cert',False) and self.config.get('enable_encrypted_communication',False):
+                        self.tls_set(ca_certs=certs.get('broker_ca_cert',None),
+                                      certfile=None,
+                                      keyfile=None,
+                                      cert_reqs=ssl.CERT_NONE)
+
+                        self.tls_insecure_set(True)
+
+                        logger.verbose('Basic SSL/TLS encrypted communications enabled')
+                        logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+                    #Handle SSL/TLS communication with certificate authentication
+                    if self.config.get('certificates',False):
+                        certs = self.config.get('certificates',{})
+                        logger.debug('TLS certificate config: {}'.format(str(certs)))
+
+                        tls_version_map = {'tlsv1.2': ssl.PROTOCOL_TLSv1_2, 'tlsv1.1': ssl.PROTOCOL_TLSv1_1, 'tlsv1': ssl.PROTOCOL_TLSv1, '': None}
+                        tls_version = tls_version_map.get(certs.get('tls_version'),ssl.PROTOCOL_TLSv1_2)
+                        logger.verbose('TLS version is {}'.format(str(tls_version)))
+
+                        if certs.get('validate_broker_cert',True):
+                            cert_required = ssl.CERT_REQUIRED
+                        else:
+                            cert_required = ssl.CERT_NONE
+                        logger.verbose('TLS cert_required={}'.format(str(cert_required)))
+
+                        self.tls_set(ca_certs=certs.get('broker_ca_cert',None),
+                                      certfile=certs.get('client_cert',None),
+                                      keyfile=certs.get('client_key',None),
+                                      cert_reqs=cert_required,
+                                      tls_version=tls_version)
+
+                        if not certs.get('validate_broker_cert',None):
+                            self.tls_insecure_set(True)
+
+                            logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+                        else:
+                            logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+                    #Handle user/pass authentication
+                    if self.config.get('username',False) or self.config.get('password',False):
+                        logger.verbose('Credential passwords s are redacted to protect the innocent...')
```

Comment:
verbose seems too high for this IMO. Also, best calculate the variable once since you seem to reuse it

Reply:
Moved logging level down to debug in commit b7881f7e458811f267deb40f9d571826fd759525

I don't think I'm actually reusing anything - I display hash of None and actual password for which code looks similar, but isn't same

Reply:
Gotcha